cli/azd/pkg/tools/git/git.go

```diff
@@ -245,3 +248,58 @@ func (cli *gitCli) IsUntrackedFile(ctx context.Context, repositoryPath string, f
 
 	return false, nil
 }
+
+// SetGitHubAuthForRepo creates git config for the repositoryPath like
+//
+// [credential "https://github.com"]  (when credential is equal to "https://github.com")
+// helper =
+// helper = !ghPath auth git-credential
+//
+// This way, git commands run from repositoryPath will use gh as auth credential.
+// Note: Removes any previous configuration for the credential.
+// Note: `helper = ` is intentional to break the chain of previously configured global helpers.
+// See: https://github.com/cli/cli/issues/3796 for more about this strategy.
+func (cli *gitCli) SetGitHubAuthForRepo(ctx context.Context, repositoryPath, credential, ghPath string) error {
```

Comment:
Wouldn't this be already done usually during `gh auth login`?

If we are going down this path, can we leverage `gh auth setup-git`? https://github.com/cli/cli/pull/4246